org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/InitHandler.java

```diff
@@ -160,6 +178,10 @@ public IStatus runInWorkspace(IProgressMonitor monitor) {
 				connection.sendStatus(ServiceStatus.Starting, "Init...");
 				SubMonitor subMonitor = SubMonitor.convert(new ServerStatusMonitor(), 100);
 				try {
+					IWorkspace workspace = ResourcesPlugin.getWorkspace();
+					IWorkspaceDescription description = workspace.getDescription();
+					description.setAutoBuilding(false);
+					workspace.setDescription(description);
```

Comment:
I'm not overly thrilled with playing with autobuild, but if we're gonna fo that then, please refactor that into:
projectsManager.setAutoBuilding(boolean), checking new value against current workspace value, if equal, do nothing